src/compiler/parser.ts

```diff
@@ -1971,14 +1971,8 @@ namespace Parser {
 
         // If we parsed this as an external module, it may contain top-level await
         if (!isDeclarationFile && isExternalModule(sourceFile) && sourceFile.transformFlags & TransformFlags.ContainsPossibleTopLevelAwait) {
-            const oldSourceFile = sourceFile;
             sourceFile = reparseTopLevelAwait(sourceFile);
-            if (sourceFile !== oldSourceFile) {
-                setFields(sourceFile);
-                // While we are parsing the file, we dont want to use the prev source file as original node anywhere
-                // as it wont have information that gets set on the source file later by program (like paths, computed imports etc)
-                sourceFile.original = undefined;
-            }
+            setFields(sourceFile);

```

Comment:
this should probably still be behind the condition since there's no reason to set fields again if nothing changed.